Add Overwatch League skins to each hero

TheTural opened the pull request:
Adds OWL skins to each hero's page. Calculates totals correctly and has the skins list collapsed by default (because it's a long list and looks bad to be always on the page).

I've also included a python script I wrote that adds the necessary data directly into master.json - It would definitely be better to just handle the generation of owlskins info when doing the initial parsing of game data, but I didn't have a rawData.txt to work with and didn't know where to get one, so I wrote the python script as a brute force way of adding the data to master.json

I have a fork of this that is live here for viewing/testing: https://thetural.github.io/

TheTural committed:
Make OWL section collapsible

TheTural committed:
Add OWL item type

TheTural committed:
Fix item total counters for OWL items

TheTural committed:
Add OWL skins to data JSON

TheTural committed:
Fixed index page total count

TheTural committed:
Added skin requirement to conditions

Fixes issue with OWL sprays being not counted due to matching any OWL item

TheTural committed:
Fixed skin qualifier

TheTural committed:
Changed JSON output name

TheTural committed:
My JSON backup with OWL items

TheTural committed:
Merge branch 'pr/1' into development

Merge recent changes from main repo

TheTural committed:
Add OWLSkins type

TheTural committed:
Add OWL Skins to master.json

TheTural committed:
Script to add OWL Skins to data file

Python script that loops through heroes and adds all the teams' OWL skins to their data set in master.json -teamsList array will need updated when new teams are added in Season 2

TheTural committed:
Ignore the backup json my script creates

TheTural committed:
Revert "Add OWLSkins type"

This reverts commit f86457131e0cf372fbb06db642a82c9c45cea403.

TheTural committed:
Get rid of my own cosmetics data

TheTural committed:
Remove my name from the json export

TheTural commented:
I just realized that this makes some superfluous changes. Not sure how that happened, but probably due to my github incompetence. I merged the current Master branch to my fork, then did the PR off of that.

These don't need changes from me:
css/_global.scss
css/main.min.css
data/events.json
helpers/dataMapper/EVENTDATA.js

Js41637 commented:
Nice work putting this together, I'll likely pull this locally and use some of the work you've done and make some changes.

I gotta be careful with OWL stuff because because there are 12 teams + the OWL Team, then there is away team versions of every skin, soon there will also be contenders skins + away. Not to mention there will also be 6 new OWL teams next year.

I mostly wanna think about how I want the data to be structured, adding 28 (34 next year) skins to every hero is just a massive bloat to the json when they're essentially all the same.

Gonzom commented:
My suggestion would be to put all the OWL league skins in their own page (like the event ones), divided either by teams or by heroes. When one of those skins is marked as owned, the hero's page would update, showing it in the list and updating the value. If it can't be shown in the list, then just update the value.

TheTural committed:
All Stars skins

TheTural committed:
D.Va Nano items

TheTural committed:
Fix bad copy and paste job

TheTural committed:
More D.Va Nano stuff

TheTural committed:
Syncing stuff with web page version

TheTural committed:
D.Va Nano item descriptions

TheTural committed:
Remove exclamation point

TheTural committed:
Fix All Star icon desc

TheTural committed:
Lucioball spray/icon fixes

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

Merging Halloween changes

TheTural committed:
Re-add OWL skins

TheTural committed:
Add All Stars skins

TheTural committed:
Make header clickable

Wanted to be able to go back to the home page without clicking into a sub-menu

TheTural committed:
OWL Skin Previews (Ana)

TheTural committed:
OWL Skins Previews

TheTural committed:
OWL skins preview stuff

TheTural committed:
Ashe additions

TheTural committed:
Fix encoding on data file

TheTural committed:
Other encoding fixes

TheTural committed:
OWL Away skins pictures

TheTural committed:
Ashe consistency update

Change colors to match what Js41637 uses and added his images

TheTural committed:
Typo

TheTural committed:
Add files via upload

TheTural committed:
Merge branch 'master' into development

TheTural committed:
Create master.json

TheTural committed:
Update master.json

TheTural committed:
Merge branch 'development' of https://github.com/TheTural/Overwatch-Item-Tracker into development

TheTural committed:
OWL Skins

TheTural committed:
Update master.json

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

TheTural committed:
OWL skins

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

TheTural committed:
OWL skins

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

TheTural committed:
OWL skins

Also adding new teams because I forgot to last patch

TheTural committed:
New OWL Skins

TheTural committed:
Baptiste

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

TheTural committed:
OWL

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

TheTural committed:
OWL skins

TheTural committed:
All Stars 2019

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

TheTural committed:
OWL Skins

Add OWL skins to data, also remove the All Stars skins from the OWL category/script because they're included in js41637's master.json now

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

TheTural committed:
Update master.json

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

TheTural committed:
OWL

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

TheTural committed:
OWL

TheTural committed:
Lego and OWL

Add Bastion Brick challenge and OWL season 1 Winston and Zen skins

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

TheTural committed:
Update owlSkinAdder.py

TheTural committed:
Update master.json

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

TheTural committed:
Update master.json

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

TheTural committed:
OWL

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

TheTural committed:
Update master.json

TheTural committed:
OWL Legendary skin fix

Remove Winston and Zen skins from the OWL section since they're in the main section now

TheTural committed:
Add some missing OWL skin screenshots

TheTural committed:
Screenshotter script fix

TheTural committed:
Merge remote-tracking branch 'upstream/development' into development

TheTural committed:
Update master.json
